### PR TITLE
Initial update for supporting play 2.5.x

### PR DESCRIPTION
**Pull request by GregBowyer**
Any of the checked boxes below indicate that I took action:

- [x] Reviewed the [Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md#contribution-workflow).
- [x] Signed the [Gradle CLA](http://gradle.org/contributor-license-agreement/).
- [x] Ensured that basic checks pass: `./gradlew quickCheck`

For all non-trivial changes that modify the behavior or public API:

- [ ] Before submitting a pull request, I started a discussion on the
[Gradle developer list](https://groups.google.com/forum/#!forum/gradle-dev)
or can reference a [JIRA issue](https://issues.gradle.org/secure/Dashboard.jspa).
- [ ] I wrote a design document. A design document can be
brief but explains the use case or problem you are trying to solve,
touches on the planned implementation approach as well as the test cases
that verify the behavior. Optimally, design documents should be submitted
as a separate pull request. [Samples](https://github.com/gradle/gradle/tree/master/design-docs)
can be found in the Gradle GitHub repository. Please let us know if you need help with
creating the design document. We are happy to help!
- [ ] The pull request contains an appropriate level of unit and integration
test coverage to verify the behavior. Before submitting the pull request
I ran a build on your local machine via the command
`./gradlew quickCheck <impacted-subproject>:check`.
- [ ] The pull request updates the Gradle documentation like user guide,
DSL reference and Javadocs where applicable.

### CODE REVIEW

**Comment by abiegert** [comment]
Our development team is currently using play framework 2.5.4 as part of a larger enterprise project. We would very much like to use gradle instead of sbt to build the project., However the current gradle version only support play version 2.4.x.

Any chance that the gradle team could give an estimate when the play 2.5.x support will be available in an official release?

Thanks in advance for your help!

File: subprojects/platform-play/src/test/groovy/org/gradle/play/internal/PlayPlatformResolverTest.groovy [hunk 1]
```diff
@@ -78,6 +78,34 @@ class PlayPlatformResolverTest extends Specification {
...
+
+        then:
+        playPlatform.name == "play-2.5.4"
+        playPlatform.playVersion == "2.5.4"
+        playPlatform.javaPlatform.targetCompatibility == JavaVersion.current()
+        playPlatform.scalaPlatform.scalaVersion == "2.11.4"
+
+        where:
+        requirement << ["play-2.5.4", [play: '2.5.4']]
+    }
```
Comment (wolfs) [inline subprojects/platform-play/src/test/groovy/org/gradle/play/internal/PlayPlatformResolverTest.groovy]: Please collapse these to tests into one. You should be able to do that by adjusting the where class.

**Comment by wolfs** [comment]
Thank you very much for your pull request!

I added a line comment to your commit. Please address this one.

**Comment by wolfs** [comment]
Please update the section in the userguide, too.

**Comment by GregBowyer** [comment]
Ok userguide and unit test updated

File: subprojects/platform-play/src/test/groovy/org/gradle/play/internal/PlayPlatformResolverTest.groovy [hunk 2]
```diff
@@ -44,49 +44,38 @@ class PlayPlatformResolverTest extends Specification {
...
+        playPlatform.scalaPlatform.scalaVersion == scalaPlatform
 
         where:
-        requirement << ["play-2.2.3", [play: '2.2.3']]
+        requirement       | playVersion | scalaPlatform
+        "play-2.2.3"      | "2.2.3"     | "2.10.4"
+        "play-2.3.4"      | "2.3.4"     | "2.11.4"
+        "play-2.4.6"      | "2.4.6"     | "2.11.4"
+        "play-2.5.4"      | "2.5.4"     | "2.11.4"
+        "play-2.3.1-2.10" | "2.3.1"     | "2.10.4"
```
Comment (wolfs) [inline subprojects/platform-play/src/test/groovy/org/gradle/play/internal/PlayPlatformResolverTest.groovy]: This test fails for `play-2.3.1-2.10` - please fix it. Requirement should be - as before - sometimes a `Map` and sometimes a `String`. If it gets to complicated feel free to split this tests to multiple tests again.

**Comment by wolfs** [comment]
Thanks for the update. We now have a failing test. Could you please fix it?

**Comment by GregBowyer** [comment]
Fixed the failing unit test

**Comment by wolfs** [comment]
Thank you for your work! The pull request has been merged an will be included in Gradle 3.1.

**Comment by wolfs** [comment]
Internal review: https://code-review.gradle.org/cru/REVIEW-6135

**Comment by eriwen** [comment]
Nice work @GregBowyer!

**Comment by GregBowyer** [comment]
Thanks for the merge

**Pull request merged**